lib/sqlalchemy/dialects/postgresql/types.py

```diff
@@ -247,6 +250,17 @@ def as_generic(self, allow_nulltype: bool = False) -> sqltypes.Interval:
     def python_type(self) -> Type[dt.timedelta]:
         return dt.timedelta
 
+    @no_type_check
+    def literal_processor(
+        self, dialect: Optional[Dialect]
+    ) -> Callable[[Optional[dt.timedelta]], Optional[str]]:
+        def process(value: Optional[dt.timedelta]) -> Optional[str]:
+            if value is not None:
+                return f"make_interval(secs=>{value.total_seconds()})"
+            return None
```

Comment:
**mike bayer** (zzzeek) wrote:

this would return the string "NULL".  we dont typically implement or test None for the literal processors since I dont know that's too much of a real use case, ideally we'd address it all at once by having the compiler not actually call literal processors when the value is NULL in any case.  This does get into the issue of JSON "null" etc. but we have flags for that.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4873